app/controllers/UploadYourFilesReceiptController.scala

```diff
@@ -35,20 +36,36 @@ import scala.concurrent.{ExecutionContext, Future}
 class UploadYourFilesReceiptController @Inject()(
   authenticate: AuthAction,
   requireEori: EORIRequiredAction,
-  getData: DataRetrievalAction,
-  requireResponse: FileUploadResponseRequiredAction,
   cdsFileUploadConnector: CdsFileUploadConnector,
   metrics: SfusMetrics,
-  uploadYourFilesReceipt: upload_your_files_receipt
+  uploadYourFilesReceipt: upload_your_files_receipt,
+  answersConnector: AnswersConnector
 )(implicit mcc: MessagesControllerComponents, ec: ExecutionContext)
     extends FrontendController(mcc) with I18nSupport {
 
-  def onPageLoad(): Action[AnyContent] = (authenticate andThen requireEori andThen getData andThen requireResponse).async { implicit req =>
-    addFilenames(req.fileUploadResponse.uploads).map { uploads =>
-      Ok(uploadYourFilesReceipt(uploads, req.userAnswers.mrn))
+  def onPageLoad(): Action[AnyContent] = (authenticate andThen requireEori).async { implicit req =>
+    answersConnector.findByEori(req.eori).flatMap { maybeUserAnswers =>
```

Comment:
I am wondering why is this logic different frmo what is done in Movements services?
Here, cache is cleared on GET endpoint for confirmation page, while in the other services it is always done on POST endpoint for Summary page. This allows clearing cache on every page refresh, but I am not sure if this is a problem.

What worries me, is that the approach taken here is different from other services. For consistency purposes I'd rather change this implementation, unless there is some kind of limitation.
To pass data from the previous page, you can use Flash cookie. This is exactly what's being done for Movements.

Reply:
As you mentioned on a call, the data required on Confirmation Page is not a simple `String` value, but rather the whole object (for files).
Therefore, it makes more sense to keep data in cache up to reaching the Confirmation Page.